File: magma_scripts/new-cd8-cells/original_with_trb_zero_char.m
SetNthreads(64);
Q := RationalField();
SetVerbose("Faugere", 2);
P<N_6, M_6, S_5, P_5, N_5, M_5, E_5, S_4, P_4, N_4, M_4, E_4, S_3, P_3, N_3, M_3, E_3, S_2, P_2, N_2, M_2, E_2, S_1, P_1, N_1, M_1, E_1, S_0, P_0, N_0, M_0, E_0, z_aux, w_aux, delta_EL, delta_LM, delta_NE, mu_EE, mu_LE, mu_LL, mu_M, mu_N, mu_P, mu_PE, mu_PL, rho_E, rho_P>:= PolynomialRing(Q, 47, "grevlex");
G := ideal< P | 177030516317694458-N_0, N_0*P_0*delta_NE+N_0*mu_N+N_1, 223808830952811077-M_0, M_0*mu_M-S_0*delta_LM+M_1, 67945318210523582-S_0-E_0, E_0^2*mu_EE-E_0*P_0*rho_E-N_0*P_0*delta_NE+E_0*delta_EL+E_1, E_0*S_0*mu_LE+S_0^2*mu_LL-S_0*delta_EL+S_0*delta_LM+S_1, -N_1-4541274779094086634124857312927138707788674424182668, (P_0*delta_NE+mu_N)*N_1+N_2+N_0*delta_NE*P_1, E_0*P_0*mu_PE-P_0^2*rho_P+P_0*S_0*mu_PL+P_0*mu_P+P_1, -M_1-23436606432473867985552756153093407, M_1*mu_M-S_1*delta_LM+M_2, -E_1-S_1+4825007986318108591582658182637481471521843649558608, (2*E_0*mu_EE-P_0*rho_E+delta_EL)*E_1+(-N_0*P_1-N_1*P_0)*delta_NE-E_0*P_1*rho_E+E_2, S_0*mu_LE*E_1+(E_0*mu_LE+2*S_0*mu_LL-delta_EL+delta_LM)*S_1+S_2, -N_2+138975406858849841223897426887299414235821207288947326888948595631570818600721339580220, (N_0*P_2+2*N_1*P_1+N_2*P_0)*delta_NE+N_2*mu_N+N_3, (E_0*mu_PE-2*P_0*rho_P+S_0*mu_PL+mu_P)*P_1+(E_1*mu_PE+S_1*mu_PL)*P_0+P_2, -M_2-53125208079861290068452281838847441973990717902076129614722005477707, M_2*mu_M-S_2*delta_LM+M_3, -E_2-S_2-32165498095148370584213551486224321677076505906637037148592863518850350657138311860888, (-N_0*P_2-2*N_1*P_1-N_2*P_0)*delta_NE+(-E_0*P_2-2*E_1*P_1-E_2*P_0)*rho_E+(2*E_0*mu_EE+delta_EL)*E_2+2*E_1^2*mu_EE+E_3, (E_0*mu_LE+2*S_0*mu_LL-delta_EL+delta_LM)*S_2+(2*E_1*S_1+E_2*S_0)*mu_LE+2*S_1^2*mu_LL+S_3, -N_3-1686862491020156054957304588593012401267721820490310090605720396736202568039800659054929856179549109188741418659240720592, (N_0*P_3+3*N_1*P_2+3*N_2*P_1+N_3*P_0)*delta_NE+N_3*mu_N+N_4, (E_0*mu_PE-2*P_0*rho_P+S_0*mu_PL+mu_P)*P_2+(2*E_1*P_1+E_2*P_0)*mu_PE+(P_0*S_2+2*P_1*S_1)*mu_PL-2*P_1^2*rho_P+P_3, -M_3-4595462879606097679830923841599837721730251567409897940614098017453806062881423342867775010480500166847, M_3*mu_M-S_3*delta_LM+M_4, -E_3-S_3-11775829027889779358044284465892601875382097586390804697680919129832996034840830754134134220070209824472598282913977254868, (-N_0*P_3-3*N_1*P_2-3*N_2*P_1-N_3*P_0)*delta_NE+(-E_0*P_3-3*E_1*P_2-3*E_2*P_1-E_3*P_0)*rho_E+(2*E_0*mu_EE+delta_EL)*E_3+6*E_1*E_2*mu_EE+E_4, (E_0*S_3+3*E_1*S_2+3*E_2*S_1+E_3*S_0)*mu_LE+(2*S_0*mu_LL-delta_EL+delta_LM)*S_3+6*S_1*S_2*mu_LL+S_4, -N_4-185715037968575199303449862227344113426689763330626980399988640192346570191829765940384107905023517121321284358544729100106300258723791989328850825355536240, (N_0*P_4+4*N_1*P_3+6*N_2*P_2+4*N_3*P_1+N_4*P_0)*delta_NE+N_4*mu_N+N_5, (E_0*P_3+3*E_1*P_2+3*E_2*P_1+E_3*P_0)*mu_PE+(P_0*S_3+3*P_1*S_2+3*P_2*S_1+P_3*S_0)*mu_PL+(-2*P_0*rho_P+mu_P)*P_3-6*P_1*P_2*rho_P+P_4, -M_4+188665600837858690024759067791047587321263010880369071645259679083758069924981367190678475010137168699776952463923130206020869660016623125, M_4*mu_M-S_4*delta_LM+M_5, -E_4-S_4-474019461312725854457170193867724149312216212134730178428340693469944886658460084580242361780067741102770490121633738724866865042969314714660492859039854868, (-N_0*P_4-4*N_1*P_3-6*N_2*P_2-4*N_3*P_1-N_4*P_0)*delta_NE+(-E_0*P_4-4*E_1*P_3-6*E_2*P_2-4*E_3*P_1-E_4*P_0)*rho_E+(2*E_0*E_4+8*E_1*E_3+6*E_2^2)*mu_EE+E_4*delta_EL+E_5, (E_0*S_4+4*E_1*S_3+6*E_2*S_2+4*E_3*S_1+E_4*S_0)*mu_LE+(2*S_0*mu_LL-delta_EL+delta_LM)*S_4+(8*S_1*S_3+6*S_2^2)*mu_LL+S_5, -N_5+9325699641403896183296134265810890558811517968262113459508965180949683563156649165076661984818732140000811640717003130596875452505832452514269276886692188125281234791336312972560985265496544, (N_0*P_5+5*N_1*P_4+10*N_2*P_3+10*N_3*P_2+5*N_4*P_1+N_5*P_0)*delta_NE+N_5*mu_N+N_6, (E_0*P_4+4*E_1*P_3+6*E_2*P_2+4*E_3*P_1+E_4*P_0)*mu_PE+(P_0*S_4+4*P_1*S_3+6*P_2*S_2+4*P_3*S_1+P_4*S_0)*mu_PL+(-2*P_0*rho_P+mu_P)*P_4+(-8*P_1*P_3-6*P_2^2)*rho_P+P_5, -M_5+25739552448164877561384212126855421163211568007478693704939748399619520372794453537966790409127186144446824280118698664782274950863497111580805686326522626643664001656671841, M_5*mu_M-S_5*delta_LM+M_6, -N_6+733839626541838023615071047924848362921564833984456198526097597079698672914403440485419314485250705012879823773807219705247145951585464498406614766193845967689579179420663867437448367782502667334066294221648468023533356407008, -M_6-1248403903991535433213390276968461322141418376442176234055825526826930209672252561726503108252700316016711692781394252169560539287254123339135391417440411542418178587783345644403823890600232621659484577240619, -E_5-S_5+109715924412137010413903568915594591596123520510312889941982816726748232763369246409538161302623346723848948282704708492767362853258925674389475590622981329043758555001353948098545868935449572, z_aux-1>;
time GroebnerBasis(G);// {}
quit;

File: julia_scripts/original_hiv2/original_with_trb_zero_char.jl
using Oscar
R, vars = PolynomialRing(QQ,"w_9, z_8, y_8, w_8, z_7, y_7, x_7, w_7, v_7, z_6, y_6, x_6, w_6, v_6, z_5, y_5, x_5, w_5, v_5, z_4, y_4, x_4, w_4, v_4, z_3, y_3, x_3, w_3, v_3, z_2, y_2, x_2, w_2, v_2, z_1, y_1, x_1, w_1, v_1, z_0, y_0, x_0, w_0, v_0, z_aux, w_aux, a, b, beta, c, d, h, k, lm, q, u")
I = ideal(R, [4783447931548041900268207-w_0, c*q*w_0*y_0-c*w_0*y_0*z_0+b*w_0+w_1, 11079185645591880054754775-z_0, -c*q*w_0*y_0+h*z_0+z_1, -w_1+558890367203377890630485292554533408203626388732360422400317121622504207095682736608624276426198876, ((-z_1*y_0+y_1*(q-z_0))*w_0+w_1*y_0*(q-z_0))*c+b*w_1+w_2, -beta*v_0*x_0+a*y_0+y_1, 567553512025606026074914733959898708908354036403800501058161933207412255864553214556053875615015700-z_1, -q*(w_0*y_1+w_1*y_0)*c+h*z_1+z_2, -w_2+123004199732229301806589219287187294855042193030314333418944342604256609990122405363395312955689227171312752318671846813380425679396373421492001997315223900830625532394257968, ((-z_2*y_0-2*z_1*y_1-y_2*(z_0-q))*w_0+(-2*z_1*w_1-w_2*(z_0-q))*y_0-2*w_1*y_1*(z_0-q))*c+b*w_2+w_3, -beta*v_0*x_1-beta*v_1*x_0+a*y_1+y_2, -k*y_0+u*v_0+v_1, beta*v_0*x_0+d*x_0-lm+x_1, 66312040035293931785145288002521037390463332501341632633631551784093535324443752854194573430990429062461362141424175192322149593367906472722525855302224482653494901123420600-z_2, -q*(w_0*y_2+2*w_1*y_1+w_2*y_0)*c+h*z_2+z_3, -w_3+34597862630316879224206378612511705354279877660573879986878999930290959804358130082887614421350695036039431081143086674726353723516972234996453038544530361887176982082349994367777344660700514953057675868108062299108015712134443233587188487412618624, ((q*y_3-y_0*z_3-3*y_1*z_2-3*y_2*z_1-y_3*z_0)*w_0+(q*w_3-3*w_1*z_2-3*w_2*z_1-w_3*z_0)*y_0+(-6*z_1*y_1-3*y_2*(z_0-q))*w_1-3*y_1*w_2*(z_0-q))*c+b*w_3+w_4, (-v_0*x_2-2*v_1*x_1-v_2*x_0)*beta+a*y_2+y_3, -k*y_1+u*v_1+v_2, beta*x_0*v_1+(beta*v_0+d)*x_1+x_2, 14594381824771558452408201197140644313800825670177234349181759905944064171932937254510723315884096004412717016508589189064377699736135949671859289769207212536347451471881236420324525414589027921498908360679197029119799380506315807381347997955753200-z_3, -3*q*(y_2*w_1+1/3*y_3*w_0+w_2*y_1+1/3*w_3*y_0)*c+h*z_3+z_4, -w_4+12340925936378144838602350287416591645644631691244169514524155877198056895883069372520131518050973752985610555784876538365302208259249188704659423341218352787602432774104810764177857288998210349987526824729014164293465017987169034931963982715037021127019834024103369242941626682451816978471361424372116314447467430144104832, ((q*y_4-y_0*z_4-4*y_1*z_3-6*y_2*z_2-4*y_3*z_1-y_4*z_0)*w_0+(q*w_4-4*w_1*z_3-6*w_2*z_2-4*w_3*z_1-w_4*z_0)*y_0+(4*q*y_3-12*y_1*z_2-12*y_2*z_1-4*y_3*z_0)*w_1+(4*q*w_3-12*w_2*z_1-4*w_3*z_0)*y_1-6*y_2*w_2*(z_0-q))*c+b*w_4+w_5, (-v_0*x_3-3*v_1*x_2-3*v_2*x_1-v_3*x_0)*beta+a*y_3+y_4, -k*y_2+u*v_2+v_3, (v_0*x_2+2*v_1*x_1+v_2*x_0)*beta+d*x_2+x_3, -w_5+5262418647824475882224138380767025586095272523088327411750216025726134758043539505076119302629957664751122809615014267354322000580842947377360567542397538496138711569390884538067427266762314215358227364897121688376753323373236527371177367159014840161820625082251680430602322354908780145401229223751649780998159130109399849384564209922211023017594081138115349919966737998849717370007431597868718976, ((q*y_5-y_0*z_5-5*y_1*z_4-10*y_2*z_3-10*y_3*z_2-5*y_4*z_1-y_5*z_0)*w_0+(q*w_5-5*w_1*z_4-10*w_2*z_3-10*w_3*z_2-5*w_4*z_1-w_5*z_0)*y_0+(5*q*y_4-20*y_1*z_3-30*y_2*z_2-20*y_3*z_1-5*y_4*z_0)*w_1+(5*q*w_4-30*w_2*z_2-20*w_3*z_1-5*w_4*z_0)*y_1+(-10*w_2*y_3-10*w_3*y_2)*z_0+(10*w_2*y_3+10*w_3*y_2)*q-30*y_2*z_1*w_2)*c+b*w_5+w_6, (-v_0*x_4-4*v_1*x_3-6*v_2*x_2-4*v_3*x_1-v_4*x_0)*beta+a*y_4+y_5, -4*q*(y_1*w_3+3/2*y_2*w_2+y_3*w_1+1/4*y_4*w_0+1/4*w_4*y_0)*c+h*z_4+z_5, -k*y_3+u*v_3+v_4, (v_0*x_3+3*v_1*x_2+3*v_2*x_1+v_3*x_0)*beta+d*x_3+x_4, -w_6+2621117188753010330885184088661008020112320814900909567890058496816703975859961971725736536789281246792896265482448664597171926116158156266159754951981753169034028614555093083980751884619863227289555106991473647160889564097485910176826026542450245748924639965331992791388175015313180665733191272234445843786913861721410303819431030283382488196121819580842182515927986253034309773807755030118805234306522739098279236512399500843868267713071142960234220078974741329032987968, ((q*y_6-y_0*z_6-6*y_1*z_5-15*y_2*z_4-20*y_3*z_3-15*y_4*z_2-6*y_5*z_1-y_6*z_0)*w_0+(q*w_6-6*w_1*z_5-15*w_2*z_4-20*w_3*z_3-15*w_4*z_2-6*w_5*z_1-w_6*z_0)*y_0+(6*q*y_5-30*y_1*z_4-60*y_2*z_3-60*y_3*z_2-30*y_4*z_1-6*y_5*z_0)*w_1+(6*q*w_5-60*w_2*z_3-60*w_3*z_2-30*w_4*z_1-6*w_5*z_0)*y_1+(-15*w_2*y_4-20*w_3*y_3-15*w_4*y_2)*z_0+(15*w_2*y_4+20*w_3*y_3+15*w_4*y_2)*q+(-90*y_2*z_2-60*y_3*z_1)*w_2-60*y_2*z_1*w_3)*c+b*w_6+w_7, (-v_0*x_5-5*v_1*x_4-10*v_2*x_3-10*v_3*x_2-5*v_4*x_1-v_5*x_0)*beta+a*y_5+y_6, -q*(w_0*y_5+5*w_1*y_4+10*w_2*y_3+10*w_3*y_2+5*w_4*y_1+w_5*y_0)*c+h*z_5+z_6, -k*y_4+u*v_4+v_5, (v_0*x_4+4*v_1*x_3+6*v_2*x_2+4*v_3*x_1+v_4*x_0)*beta+d*x_4+x_5, -w_7+1491531531788152217074401245015793770826269165076958468884731306108864514450704796665192208308414994830694123431136354090308060463713283908945814052000310783212076738154328224211206958496074369702913525935518966935873686793885744640467850941920304947925487773862759181987771232847681609907148755103413331223401543466632458066085043914092422965498024493193772195019923493935972150993095437766412273831747956229355058590571169144971637571629829027648900385495230239676474476630123205367952021075580651175715428791517597554511274168843880584688037824, ((q*y_7-y_0*z_7-7*y_1*z_6-21*y_2*z_5-35*y_3*z_4-35*y_4*z_3-21*y_5*z_2-7*y_6*z_1-y_7*z_0)*w_0+(q*w_7-7*w_1*z_6-21*w_2*z_5-35*w_3*z_4-35*w_4*z_3-21*w_5*z_2-7*w_6*z_1-w_7*z_0)*y_0+(7*q*y_6-42*y_1*z_5-105*y_2*z_4-140*y_3*z_3-105*y_4*z_2-42*y_5*z_1-7*y_6*z_0)*w_1+(7*q*w_6-105*w_2*z_4-140*w_3*z_3-105*w_4*z_2-42*w_5*z_1-7*w_6*z_0)*y_1+(-21*w_2*y_5-35*w_3*y_4-35*w_4*y_3-21*w_5*y_2)*z_0+(21*w_2*y_5+35*w_3*y_4+35*w_4*y_3+21*w_5*y_2)*q+(-210*y_2*z_3-210*y_3*z_2-105*y_4*z_1)*w_2+(-210*w_3*z_2-105*w_4*z_1)*y_2-140*y_3*z_1*w_3)*c+b*w_7+w_8, (-v_0*x_6-6*v_1*x_5-15*v_2*x_4-20*v_3*x_3-15*v_4*x_2-6*v_5*x_1-v_6*x_0)*beta+a*y_6+y_7, -q*(w_0*y_6+6*w_1*y_5+15*w_2*y_4+20*w_3*y_3+15*w_4*y_2+6*w_5*y_1+w_6*y_0)*c+h*z_6+z_7, -k*y_5+u*v_5+v_6, (v_0*x_5+5*v_1*x_4+10*v_2*x_3+10*v_3*x_2+5*v_4*x_1+v_5*x_0)*beta+d*x_5+x_6, -w_8+954939059715468859710646343346010604187994428346606372009254627756107222137761286062051801584397610439214859127964501298411005131332669042533394494485429172726313266089181583304004858171967778476118824565678290655294349057140380964536960903335844779688092759165952989409280047953213303978672782637675749834751233192558518232502437367558573284076534282085441680028629155318712809779322338241472980043121161100276408502137023260892876553685624274511715896414366076410871419496459773636814148367135926199311332667445867747254883959413408010831797258163854501461202648303299997897141343541960854919806575312595157317958744832, ((q*y_8-y_0*z_8-8*y_1*z_7-28*y_2*z_6-56*y_3*z_5-70*y_4*z_4-56*y_5*z_3-28*y_6*z_2-8*y_7*z_1-y_8*z_0)*w_0+(q*w_8-8*w_1*z_7-28*w_2*z_6-56*w_3*z_5-70*w_4*z_4-56*w_5*z_3-28*w_6*z_2-8*w_7*z_1-w_8*z_0)*y_0+(8*q*y_7-56*y_1*z_6-168*y_2*z_5-280*y_3*z_4-280*y_4*z_3-168*y_5*z_2-56*y_6*z_1-8*y_7*z_0)*w_1+(8*q*w_7-168*w_2*z_5-280*w_3*z_4-280*w_4*z_3-168*w_5*z_2-56*w_6*z_1-8*w_7*z_0)*y_1+(-28*w_2*y_6-56*w_3*y_5-70*w_4*y_4-56*w_5*y_3-28*w_6*y_2)*z_0+(28*w_2*y_6+56*w_3*y_5+70*w_4*y_4+56*w_5*y_3+28*w_6*y_2)*q+(-420*y_2*z_4-560*y_3*z_3-420*y_4*z_2-168*y_5*z_1)*w_2+(-560*w_3*z_3-420*w_4*z_2-168*w_5*z_1)*y_2+(-280*w_3*y_4-280*w_4*y_3)*z_1-560*w_3*y_3*z_2)*c+b*w_8+w_9, (-v_0*x_7-7*v_1*x_6-21*v_2*x_5-35*v_3*x_4-35*v_4*x_3-21*v_5*x_2-7*v_6*x_1-v_7*x_0)*beta+a*y_7+y_8, -7*(w_6*y_1+1/7*w_7*y_0+3*y_2*w_5+5*y_3*w_4+5*y_4*w_3+3*y_5*w_2+y_6*w_1+1/7*y_7*w_0)*q*c+h*z_7+z_8, -k*y_6+u*v_6+v_7, (v_0*x_6+6*v_1*x_5+15*v_2*x_4+20*v_3*x_3+15*v_4*x_2+6*v_5*x_1+v_6*x_0)*beta+d*x_6+x_7, -w_9+679302641070871783738432513623946727210458589755271801341562697705088693332581197339860413044776942261397365889010280799196638132311727035270261648034767323265453403949979890761444380027592800152746476673641282179482357964057145444551957587972502012699300521435314945744634712798630580115414687043647270041291843791699455712782741138172331703258877386664113327771540910371765407944869605716051957114786297845817550867895241388934307970323954369019732204801696809060509626280909194281222880488210467744791233028244182909673817541512408704362703893893423053690051222495741542912242501806181127103649724137726087934126423323973670395083454659794445964772327117647003409869529414549797090426373541376, 4105017703842984414452413366171140435784470082603012926995814177268081011477964184390058461915470905969423569277698767186407848789052857565452038209035240212661785916034276713778851775401137277873514931294746305971209978944574349755501112089412248257444374780379830653476394861887723848652598612392953162189357045224988800-z_4, 1464244193115444944613792204743334486900874264901422217584955479862109719803442985426132339542298858120759755282620807615071909986239251212734683292297685125436698422573783228190347069182966836629728444127267252710010203997336812349945338008083469876042365645223710935150690081897799691807852313367986995110197425036467666208097325508497446020115139662995924113607859336708178838989022592670600800-z_5, 624383128668289032808569469250128667416795891339221375253259628721138665347670423423081395708992197064065026277377154164833761959688283363523393995999568242457817235940161785238627414172390516302327075242226631107904776141853224413037649239122789620171665765767293386935154249271670859594162790532070753220482481317050137883988554672527394152424627546115676301315933453965856244288609809593211712118448195005829688319228037242258810155344644210007070628308549463723221200-z_6, 310994137951455102942897963630107886482664038474673104014365908371733228692086973529342917640944956554330635851847983408525143775196390945102550800070548056257247325864499719105911765912772507799375226362227235074649588353857911038869457871389760524809533206009551777972373392582729627120388273792868837761323007765216904001162403167961099721377754801698234052956293783480661648895572092291797508927047560848766033222825629241630798598807238408186828768178732834943262373335569830682998614721288842686379148151837003403472942676486428109597702800-z_7, 176969410198919331402026814031798306393113388795328542672855406005950857665136809805242541578133371333241528385023056342189254952501856328865822485527442990856591404597542340748126377598966106859431600878610781829539637195952761872833611136507009813637087057085537382977873352276148894234189007406718542713125617336573885891454714195608292440317361084240460922805984479597454816362518745141071452133173696111616539222941914639250690857712805853331918935761423526997347177736220327243126174193361138579226354223397854088795095024524558044952949853164397134040005528634036943328122112265987527921993923836343312358976745200-z_8, z_aux-1])
gb = msolve(I)
# {}